packages/xo-web/src/common/custom-fields.js

```diff
@@ -28,17 +41,31 @@ const AddCustomFieldModal = decorate([
           [name]: value,
         })
       },
+      toggleDate() {
+        const { props } = this
+        props.onChange({
+          ...props.value,
+          isDate: !props.value.isDate,
+        })
+      },
     },
   }),
   injectState,
-  ({ effects, value }) => (
+  ({ effects, value, update = false }) => (
     <Container>
+      <SingleLineRow>
+        <Col size={6}>{_('date')}</Col>
+        <Col size={6}>
+          <Toggle onChange={effects.toggleDate} value={value.isDate} disabled={update} />
```

Comment:
I don't think we need to disable this one. The use may want to switch from a date to a text field and vice versa.